src/lib/include/xtgeo/logging.hpp

```diff
@@ -0,0 +1,84 @@
+#ifndef XTGEO_LOGGING_HPP_
+#define XTGEO_LOGGING_HPP_
+#include <pybind11/pybind11.h>
+
+#include <pybind11/stl.h>
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+namespace py = pybind11;
+
+class Logger
+{
+public:
+    explicit Logger(const std::string &name)
+      : logger_name(name)
+    {
+        py::gil_scoped_acquire acquire;  // Acquire GIL (Global Interpreter Lock)
+        py::object logging = py::module::import("logging");
+        logger = logging.attr("getLogger")(logger_name);
+    }
+
+    template<typename... Args>
+    void log(const std::string &level, const std::string &message, Args... args)
+    {
+        py::gil_scoped_acquire acquire;  // Acquire GIL (Global Interpreter Lock)
+        std::ostringstream oss;
+        ((oss << args << " "),
+         ...);  // Fold expression to concatenate arguments with spaces
+        std::string formatted_message =
+          "[C++: " + logger_name + "] " + message + oss.str();
+
+        if (level == "debug") {
+            logger.attr("debug")(formatted_message);
+        } else if (level == "info") {
+            logger.attr("info")(formatted_message);
+        } else if (level == "warning") {
+            logger.attr("warning")(formatted_message);
+        } else if (level == "error") {
+            logger.attr("error")(formatted_message);
+        } else if (level == "critical") {
+            logger.attr("critical")(formatted_message);
+        } else {
+            logger.attr("info")(formatted_message);
+        }
```

Comment:
Do we need to release the GIL lock as well?

Reply:
It should be released automatically when out of scope, but we can tighten the scope